src/factory/load_gltf.rs

```diff
@@ -269,22 +276,106 @@ impl super::Factory {
                 let child_group = self.group();
                 item.group.add(&child_group);
                 stack.push(Item {
-                    node: clone_child(gltf, &child),
+                    node: clone_child(&gltf, &child),
                     group: child_group,
                 });
             }
 
+            node_map.insert(item.node.index(), (*item.group).clone());
             groups.push(item.group);
         }
 
         groups.swap_remove(0)
     }
 
+    /// Loads animations from glTF 2.0.
+    pub fn load_gltf_animations(
+        &mut self,
+        gltf: &Gltf,
+        node_map: &HashMap<GltfNodeIndex, Object>,
+        buffers: &gltf_importer::Buffers,
+    ) -> Vec<animation::Clip> {
+        let mut clips = Vec::new();
+        for animation in gltf.animations() {
+            let mut tracks = Vec::new();
+            let name = animation.name().map(str::to_string);
+            for channel in animation.channels() {
+                let sampler = channel.sampler();
+                let target = channel.target();
+                let node = target.node();
+                let object = match node_map.get(&node.index()) {
+                    Some(object) => object.clone(),
+                    // This animation does not correspond to any loaded node.
+                    None => continue,
+                };
+                let input = sampler.input();
+                let output = sampler.output();
+                use gltf::animation::InterpolationAlgorithm::*;
+                let interpolation = match sampler.interpolation() {
+                    Linear => animation::Interpolation::Linear,
+                    Step => animation::Interpolation::Discrete,
+                    CubicSpline => animation::Interpolation::Cubic,
+                    CatmullRomSpline => animation::Interpolation::CatmullRom,
+                };
+                use animation::{Binding, Track, Values};
+                let times: Vec<f32> = AccessorIter::new(input, buffers).collect();
+                let (binding, values) = match target.path() {
+                    gltf::animation::TrsProperty::Translation => {
+                        let values = AccessorIter::<[f32; 3]>::new(output, buffers)
+                            .map(|v| {
+                                mint::Vector3 {
+                                    x: v[0],
+                                    y: v[1],
+                                    z: v[2],
+                                }
+                            })
+                            .collect::<Vec<_>>();
+                        assert_eq!(values.len(), times.len());
+                        (Binding::Position, Values::Vector3(values))
+                    }
+                    gltf::animation::TrsProperty::Rotation => {
+                        let values = AccessorIter::<[f32; 4]>::new(output, buffers)
+                            .map(|r| {
+                                mint::Quaternion {
```

Comment:
similarly, can we use `mint::Quaternion::from`?

Reply:
I didn't want to risk it. :laughing:

Reply:
Totally get it, but we should encourage `mint` conversions and standards for the greater ~~god~~ good.